lib/assessment.rb

```diff
@@ -43,6 +59,13 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+    #I don't really understand the ARGV[0], just took from read-file.rb demo yesterday, and I don't know when to use it/if I need it here.
+    File.open(ARGV[0]) do |filename|
+      filename.reduce(&:+)
+      filename.gsub(/\n/, ' ')
+    end
```

Comment:
You've got a lot of the pieces here, they're just not together in the right order.